test/HealthChecks.AzureServiceBus.Tests/AzureServiceBusQueueMessageCountThresholdHealthCheckTests.cs

```diff
@@ -0,0 +1,290 @@
+using Azure;
+using Azure.Core;
+using Azure.Messaging.ServiceBus;
+using Azure.Messaging.ServiceBus.Administration;
+using HealthChecks.AzureServiceBus.Configuration;
+using NSubstitute;
+
+namespace HealthChecks.AzureServiceBus.Tests;
+
+public class azureservicebusqueuemessagecountthresholdhealthcheck_should
+{
+    private readonly string ConnectionString;
+    private readonly string FullyQualifiedName;
+    private readonly string TopicName;
+    private readonly string HealthCheckName = "unit-test-check";
+
+    private readonly ServiceBusClientProvider _clientProvider;
+    private readonly ServiceBusAdministrationClient _serviceBusAdministrationClient;
+    private readonly TokenCredential _tokenCredential;
+
+    public azureservicebusqueuemessagecountthresholdhealthcheck_should()
+    {
+        ConnectionString = Guid.NewGuid().ToString();
+        FullyQualifiedName = Guid.NewGuid().ToString();
+        TopicName = Guid.NewGuid().ToString();
+
+        _clientProvider = Substitute.For<ServiceBusClientProvider>();
+        _serviceBusAdministrationClient = Substitute.For<ServiceBusAdministrationClient>();
+        _tokenCredential = Substitute.For<TokenCredential>();
+
+        _clientProvider.CreateManagementClient(ConnectionString).Returns(_serviceBusAdministrationClient);
+        _clientProvider.CreateManagementClient(FullyQualifiedName, _tokenCredential).Returns(_serviceBusAdministrationClient);
+    }
+
+    [Fact]
+    public async Task can_create_client_with_connection_string()
+    {
+        using var tokenSource = new CancellationTokenSource();
+        var options = new AzureServiceBusQueueMessagesCountThresholdHealthCheckOptions(TopicName)
+        {
+            ConnectionString = ConnectionString,
+        };
+        var healthCheck = new AzureServiceBusQueueMessageCountThresholdHealthCheck(options, _clientProvider);
+        var context = new HealthCheckContext
+        {
+            Registration = new HealthCheckRegistration(HealthCheckName, healthCheck, HealthStatus.Unhealthy, null)
+        };
+
+        var actual = await healthCheck
+            .CheckHealthAsync(context, tokenSource.Token)
+            .ConfigureAwait(false);
+
+        _clientProvider
+            .Received(1)
+            .CreateManagementClient(ConnectionString);
+    }
+
+    [Fact]
+    public async Task reuses_existing_client_when_using_same_connection_string_with_different_topic()
+    {
+        // First call
+        using var tokenSource = new CancellationTokenSource();
+        var options = new AzureServiceBusQueueMessagesCountThresholdHealthCheckOptions(TopicName)
```

Comment:
topic or queue?

Reply:
I suppose copy-pasted from `azureservicebussubscriptionhealthcheck_should` all over the file.

Reply:
Yeah, rush job from my side. Sorry about that.